portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,40 +28,28 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
  
-  private List<String> comments;
-
-  @Override
-  public void init() {
-    comments = new ArrayList<>();
-    comments.add("This is a nice page.");
-    comments.add("You like engineering? I do too.");
-    comments.add("Cool website.");
-  }
-
+  private Comments comments = new Comments();
  
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String json = convertToJson(comments);
-
+    String json = new Gson().toJson(comments);
     response.setContentType("application/json;");
     response.getWriter().println(json);
   }
 
-    /**
-   * Converts a list of comments into a JSON string using manual String concatentation.
-   */
-  private String convertToJson(List<String> comments) {
-    String json = "{";
-    json += "\"Comments\": ";
-    json += "[" ;
-    json += "\"" + comments.get(0) + "\"";
-    json += ", ";
-    json += "\"" + comments.get(1) + "\"";
-    json += ", ";
-    json += "\"" + comments.get(2) + "\"";
-    json += "]";
-    json += "}";
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    getComment(request);
```

Comment:
We can improve the naming. getComment gives the impression it returns a comment but instead we add a comment. We either call it "addComment" or make it two steps:

Comment comment = getComment(request);
comments.addComment(comment);

Reply:
Okay, I will make it into two steps since it more accurately reflects the behavior.